document silent message dropping between humble and jazzy (#572)
osrf/rmw_zenoh: README.md

Note that composable nodes should *never* call `rclcpp::shutdown()`, as the composable node container will automatically do this.

For more details, see https://github.com/ros2/rmw_zenoh/issues/170.

> ### Distros earlier than Iron (e.g. Humble) are incompatible with Jazzy and later.

Reviewer: ```suggestion
### rmw_zenoh is incompatible between Humble and newer distributions. 
```

> Since Iron, ROS2 uses type hashes which means that messaging between Humble and later versions of ROS2 [is not possible](https://github.com/ros2/rmw_zenoh/issues/425). Node and topic discovery will work, but messages [will be silently dropped](https://github.com/ros2/rmw_zenoh/issues/569).

Reviewer 2: ```suggestion
Since Iron, ROS 2 uses type hashes which means that messaging between Humble and later versions of ROS 2 [is not possible](https://github.com/ros2/rmw_zenoh/issues/425). Node and topic discovery will work, but messages [will be silently dropped](https://github.com/ros2/rmw_zenoh/issues/569).
```

Author: I changed ROS2 -> ROS 2.
There is another location in the README where ROS2 is used, but I decided not to fix it because it is outside the scope of the commit.

> Since Iron, ROS 2 uses type hashes which means that messaging between Humble and later versions of ROS 2 [is not possible](https://github.com/ros2/rmw_zenoh/issues/425). Node and topic discovery will work, but messages [will be silently dropped](https://github.com/ros2/rmw_zenoh/issues/569).

Reviewer: Updating the language as other middlewares might support comms across Humble and newer distros.
```suggestion
Since Iron, ROS 2 introduced type hashes for messages and `rmw_zenoh` includes these type hashes in the Zenoh keyexpressions it constructs for data exchange. While participants will be discoverable, communication between Humble and newer distributions will fail, resulting in messages being silently dropped.

For more details, see https://github.com/ros2/rmw_zenoh/issues/569.
```
